### DISCUSSION TITLE
Should PDM manage python interpreter?

### DISCUSSION THREAD

**Discussion by Hnasar** [pdm-project/pdm#1864] [Ideas]
A new package manager similar to PDM made a splash recently https://github.com/mitsuhiko/rye (its 9 minute video is a good overview).

One notable thing it implements that PDM doesn't is the ability to handle automatically downloading missing interpreters, and provides an easy way to "pin" them for a project. It has some other unique features, like an ability to install tools into isolated environments similar to pipx.

I figured I'd mention it here in case people haven't seen. There's an interesting discussion about what the tool means for the community https://github.com/mitsuhiko/rye/discussions/6 and I figured I'd mention here to see if there's anything PDM can be inspired by. Cheers!

**Comment by pohlt** [comment]
Without knowing the implications, I would like to see this functionality in pdm. 👍🏼

**Comment by frostming** [comment]
FYI, We are planning to improve `findpython` with the ability to install a Python interpreter.

**Reply by pohlt** [reply]
What source do you plan to use for the interpreters? Same as `rye`?

**Reply by frostming** [reply]
There are existing providers that also install: `pyenv` and `asdf`, and for `rye`, that will be a new provider that installs from `python-standalone-build`, of course.

**Reply by bitfoo1** [reply]
Hello, 

thank you for this awesome package manager.
The last thing, what holds me back using it on all my projects ist the missing python version manager.

I would love to see this feature in the near future.
